packages/core/src/actions/sendTransaction.ts

```diff
@@ -58,16 +65,32 @@ export async function sendTransaction<
   config: config,
   parameters: SendTransactionParameters<config, chainId>,
 ): Promise<SendTransactionReturnType> {
-  const { account, chainId, connector, ...rest } = parameters
+  const { account, chainId, connector, gas: gas_, ...rest } = parameters
 
   const client = await getConnectorClient(config, {
     account,
     chainId,
     connector,
   })
 
+  const gas = await (async () => {
+    // Skip gas estimation if `null` is provided.
+    if (gas_ === null) return undefined
+
+    // Run gas estimation if no value is provided.
+    if (gas_ === undefined)
+      return viem_estimateGas(client, {
+        ...(rest as any),
+        chain: chainId ? { id: chainId } : null,
+      })
+
+    // Use provided gas value.
+    return gas_
+  })()
```

Comment:
Fine with it if you don't want to change it, but this seems too clever :)

Reply:
Fair, nested ternary is the go then lol

Reply:
Not too happy about the maintainability aspects of alternatives as it either hinders type inference (declare w/ explicit type + if/else) or readability (nested ternary). I feel like this approach is more readable whilst having type inference on `gas`?

Might roll with this despite the clever syntax on lines 76 and 80.

Bullish on block expressions in JS lol.

Reply:
Works for me!